pool/src/pool/health_factor.rs

```diff
@@ -88,21 +88,47 @@ impl PositionData {
             .unwrap_optimized()
     }
 
-    /// Check if the position data meets the minimum health factor, panic if not
-    pub fn require_healthy(&self, e: &Env) {
+    // Check if the position data is under a maximum health factor, panic if not
+    // Note: max must be 7 decimals
+    pub fn require_hf_under(&self, e: &Env, max: i128, error: PoolError) {
+        let min_health_factor = self
+            .scalar
+            .fixed_mul_floor(max, SCALAR_7)
+            .unwrap_optimized();
+        if self.as_health_factor() > min_health_factor {
+            panic_with_error!(e, error);
+        }
+    }
+
+    /// Check if the position data meets a minimum health factor, panic if not
+    /// Note: min must be 7 decimals
+    pub fn require_hf_over(&self, e: &Env, min: i128, error: PoolError) {
         if self.liability_base == 0 {
             return;
         }
-
-        // force user to have slightly more collateral than liabilities to prevent rounding errors
         let min_health_factor = self
             .scalar
-            .fixed_mul_floor(1_0000100, SCALAR_7)
+            .fixed_mul_floor(min, SCALAR_7)
             .unwrap_optimized();
         if self.as_health_factor() < min_health_factor {
-            panic_with_error!(e, PoolError::InvalidHf);
+            panic_with_error!(e, error);
         }
     }
```

Comment:
I really don't like that we are passing an error as a parameter.

Options:
1. remove these functions and just use `as_health_factor` and do the scaling in `user_liquidation_auction`, uncomment require_healthy
2. rename to `is_hf_under` and `is_hf_over`, and return a bool, have panics come from consumers

Reply:
If you choose to go with 2, remove the commented require_healthy fn too

Reply:
chose 2